app/models/file_depot_s3.rb

```diff
@@ -0,0 +1,54 @@
+class FileDepotS3 < FileDepot
+  include AmazonAuthenticationMixin
+  attr_accessor :s3
+
+  def self.uri_prefix
+    "s3"
+  end
+
+  def object_store?
+    true
+  end
+
+  def self.validate_settings(settings)
+     _log.debug("Validating settings [#{settings}]")
+    new(:uri => settings[:uri]).verify_credentials(nil, settings.slice(:username, :password))
+  end
+
+  private
+
+  def create_bucket(directory_path)
+    if (ssa_bucket = s3.bucket(directory_path)).exists?
```

Comment:
What does `ssa_bucket` mean here? I don't think this has anything to do with Smart State Analysis right?

Reply:
Absolutely correct.